### PR TITLE
[qt] Use QOpenGLFunctions for Qt5

### PR DESCRIPTION
**Pull request by tmpsantos**
Do not link directly with OpenGL, so we load it using Qt5. Also, always
build for OpenGL ES2 which is the compatibilty mode that should work on all
platforms supported by Qt5.

### CODE REVIEW

**Comment by tmpsantos** [comment]
@kkaefer @brunoabinader ready for review

**Comment by brunoabinader** [comment]
Can you please add more explaining on why are we adding `platform/` folders in platform-specific CMake files - ain't these supposed to be foundable already?

**Comment by tmpsantos** [comment]
I did like this to remove the ugly #ifdefs and they are not in the search path by default. I think including platform specific headers using <mbgl/foo/bar> namespace is valid.

File: platform/linux/mbgl/gl/gl_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#if MBGL_USE_GLES2
+    #define GL_GLEXT_PROTOTYPES
```
Comment (brunoabinader) [inline platform/linux/mbgl/gl/gl_impl.hpp]: We could probably move this #define out of the if as it happens in both cases.

**Pull request merged**